English and editorial review of the I-D. (#124)
IETF-TEAS-WG/actn-poi: draft-ietf-teas-actn-poi-applicability.md

  -
    name: Gabriele Galimberti
    email: ggalimbe56@gmail.com
  -

>     name: TBD

Reviewer: I need a placeholder to add a contributor before uploading the I-D on IETF datatracker

Reviewer: **[ACTN POI bi-weekly call (January 28, 2025)](https://github.com/IETF-TEAS-WG/actn-poi/wiki/minutes-2025-01-28)**

Add back the TBD contributor

Reviewer: Closed with https://github.com/IETF-TEAS-WG/actn-poi/pull/124/commits/c90c1b31e6bc25610672e6441695e982d55a4892

transport  networks�spanning IP/MPLS, optical, and microwave
technologies�is crucial to addressing customer demands for high-bandwidth
applications, such as ultra-fast  mobile broadband for 5G and fiber
connectivity services. The Abstraction and  Control of TE Networks (ACTN)
architecture and interfaces enable the automation  and efficient
operation of complex optical and IP/MPLS networks using standardized
interfaces and data models. This approach supports a broad spectrum of
network  services that can be requested by upper-layer applications,
meeting diverse  service-level requirements from a network perspective,
such as physical diversity, latency, bandwidth, and topology.

Packet Optical Integration (POI) represents an advanced application of
traffic engineering. In wide-area networks, packet networks based on the

Internet Protocol (IP), often augmented with Multiprotocol Label
Switching (MPLS) or Segment Routing  (SR), are typically implemented over

an optical transport network utilizing Dense Wavelength Division
Multiplexing (DWDM), occasionally with an optional Optical Transport
Network (OTN) layer.

In many existing network deployments, packet and optical networks are

> engineered and operated independently. This separation results in

Author: How about... 

...In many current network deployments, packet and optical networks are designed and operated independently. This separation leads to technical disparities between the underlying technologies (e.g., routers versus optical switches) and the associated methods for network engineering and planning (e.g., inter-domain peering...

> engineered and operated independently. This separation results in

Reviewer: The way I read this text is:
- cause: separation of engineering and operations between IP and optical
- consequence: different underlying technologies and methods

IMHO, it should be:
- cause: different underlying technologies and methods
- consequence: separation of engineering and operations between IP and optical

What about:

There are significant technical differences between the packet and optical technologies (e.g.,
routers versus optical switches) and their associated network engineering
and planning approaches (e.g., inter-domain peering optimization in IP
networks versus managing physical impairments in DWDM systems or
operating on vastly different time scales). Additionally, customer
requirements often differ between packet and optical networks, and it is
common for Service Providers to use different vendors for each domain. As
a result, the operation of these complex packet and optical networks is
often siloed, as each technology domain requires specialized skill sets.

As a consequence, in many existing network deployments, packet and optical networks are
engineered and operated independently.

This separation is inefficient for several reasons. First, integrating packet and optical
// see L174 onward

> engineered and operated independently. This separation results in

Reviewer 2: Fully agree with your new proposal, Italo. Thanks!

Routers (ASBR) or Area Border Router (ABR): in this document, the
generic term Border Router (BR) is used to represent either an ASBR
or an ABR.

The Multi-Domain Service Coordinator (MDSC) is responsible for
orchestrating the entire multi-domain, multi-technology network,
encompassing both packet and optical domains. A standardized interface,
the Multi-Domain Service Coordinator to Provisioning Network Controller
Interface (MPI), enables the MDSC to interact with various Provisioning
Network Controllers (O-PNCs and P-PNCs).

The MPI interface provides the MDSC with an abstracted topology,
concealing technology-specific details of the network and selectively
hiding topology information based on the chosen abstraction policy. The
level of abstraction is determined by the configuration parameters of the
P-PNC and O-PNC, such as offering potential connectivity information
between any Provider Edge (PE) and Border Router (BR) within a packet
network.

In the reference network of {{fig-ref-network}}, it is assumed that:

- The domain boundaries of the packet and optical domains are
congruent. In other words, each optical domain exclusively supports

> connectivity between IP routers within a single packet domain.;

Reviewer: ```suggestion
connectivity between IP routers within a single packet domain.
```

-  All the physical interfaces at inter-domain links are Ethernet
physical interfaces.

> Although emerging optical technologies (e.g., QSFP-DD ZR 400G) enable
> operators to deploy DWDM pluggable interfaces on IP routers, the adoption
> of these technologies remains limited. This is primarily because most
> operators are not yet prepared to manage packet and optical networks
> within a single unified domain. Consequently, a unified use case analysis
> falls outside the scope of this document.

Reviewer: **[ACTN POI bi-weekly call (January 28, 2025)](https://github.com/IETF-TEAS-WG/actn-poi/wiki/minutes-2025-01-28)**

```suggestion
Scenarios using coherent optical interfaces on the IP routers are outside the scope of this document.
```

This document analyzes scenarios in which all multi-technology IP links
supported by the optical network are intra-domain (intra-AS/intra-area),
such as PE-BR, PE-P, BR-P, and P-P IP links. Consequently, inter-domain
IP links are always single-technology connections, supported by
single-technology Ethernet links between physically adjacent IP routers.

Therefore, if inter-domain links between optical domains exist, they
would be utilized to support multi-domain optical services, which fall
outside the scope of this document.

The optical nodes within the optical domains can be either:

> - DWDM nodes, as defined in

Reviewer: I-D.ietf-ccamp-optical-impairment-topology-yang defines the 'WDM node' (not the 'DWDM node')

```suggestion
- WDM nodes, as defined in
```

There are many options to implement multi-domain L2/L3 VPNs,
including:

1. BGP-Labeled Unicast (BGP-LU) ({{?RFC8277}})

1. Inter-domain RSVP-TE

1. Inter-domain SR-TE

This document explores inter-domain TE options where the TE tunnel model,
as defined in {{!I-D.ietf-teas-yang-te}}, applies at the MPI for both
intra-domain and inter-domain TE configurations. The assessment of
alternative options is beyond the scope of this draft.

It is also assumed that:

- the bandwidth of each intra-domain TE path is managed by its
respective P-PNC;

- technology-specific mechanisms are employed for inter-domain TE path
stitching. In the case of inter-domain SR-TE, a Segment Identifier (SID)
is used in Segment Routing (SR) to define a segment (a portion of the
path) within a network. A binding SID, a special type of SID, acts as a
reference to a precomputed SR policy or path.

- each packet domain in {{fig-vpn-topo}} employs technology-specific
local protection mechanisms, such as Fast Reroute (FRR) for MPLS-TE or
Topology Independent Loop-Free Alternate (TI-LFA) for SR-TE. These
mechanisms operate with an awareness of multi-technology TE path

> properties, such as Shared Risk Link Group (SRLG).

Reviewer: **[ACTN POI bi-weekly call (January 28, 2025)](https://github.com/IETF-TEAS-WG/actn-poi/wiki/minutes-2025-01-28)**

Need to add a reference to RFC 8001

Reviewer: Closed with https://github.com/IETF-TEAS-WG/actn-poi/pull/124/commits/c90c1b31e6bc25610672e6441695e982d55a4892

paths with the same topology metric, e.g., BR21 - P24 - PE23 and
BR21 - BR22 - PE23, but with different loads. It may prefer to steer
traffic on the latter as it is less loaded.

For the purposes of this document, it is assumed that the MDSC always
provides the explicit list of all hops to the P-PNCs to set up or
modify the TE path.

{: #optical-pnc-overview}

## Optical Domain Controller and NE Functions

The optical network provides underlay connectivity services to
IP/MPLS networks. The packet and optical multi-layer coordination
is handled by the MDSC, as shown in {{fig-ref-network}}.

The O-PNC is responsible to:

- Provide the MDSC with an abstract TE topology view of its underlying
  optical network resources;

- perform single-domain local path computation when requested by
  the MDSC;

- Perform optical tunnel set up when requested by the MDSC.

The mechanisms used by the O-PNC to perform intra-domain topology
discovery and path setup are typically vendor-specific and outside
the scope of this document.

Depending on the optical network type, TE topology abstraction, path
computation, and path setup can be single-layer (either OTN or DWDM)

> or multi-layer OTN/WDM. In the latter case, multi-layer coordination

Reviewer: ```suggestion
or multi-layer OTN/DWDM. In the latter case, multi-layer coordination
```

It should also be possible to correlate information from IP and
optical layers (e.g., which port, lambda/OTSi, and direction are used

> by a specific IP service on the DWDM equipment).

Reviewer: There is no definition of 'WDM equipment' or 'DWDM equipment'

I think we can use the term 'WDM node' from I-D.ietf-ccamp-optical-impairment-topology-yang as in the rest of the document

```suggestion
by a specific IP service on the WDM node).
```

## Optical Topology Discovery

The WSON Topology Model and the Flexi-grid Topology model can be used

> to report the DWDM network topology (e.g., DWDM nodes and OMS links),

Reviewer: I-D.ietf-ccamp-optical-impairment-topology-yang defines the 'WDM node' (and not the 'DWDM node')

```suggestion
to report the DWDM network topology (e.g., WDM nodes and OMS links),
```

the following information can be abstracted and not reported at
the MPI:

- the optical parameters required for optical path computation, such
as those detailed in
{{?I-D.ietf-ccamp-optical-impairment-topology-yang}};

- the underlay OTS links and ILAs of OMS links;

- the physical connectivity between the optical transponders and the
ROADMs.

The OTN Topology Model also reports the CBR client LTPs that
terminates the cross-technology Ethernet links: once CBR client LTP is
reported for
each CBR or multi-function client interface on the optical nodes (see
sections 4.4 and 5.1 of {{?I-D.ietf-ccamp-transport-nbi-app-statement}}
for the description of multi-function
client interfaces).

The Ethernet Topology Model reports the Ethernet client LTPs that
terminate the cross-technology Ethernet links: one Ethernet client LTP is
reported
for each Ethernet or multi-function client interface on the optical
nodes.

The optical transponders and, optionally, the OTN access cards, are
abstracted at MPI by the O-PNC as Trail Termination Points (TTPs),
defined in {{!RFC8795}}, within the optical network topology. This
abstraction is valid independently of the fact that optical

> transponders are physically integrated within the same DWDM node or
> are physically located on a device external to the DWDM node since it
> both cases the optical transponders and the DWDM node are under the

Reviewer: I-D.ietf-ccamp-optical-impairment-topology-yang defines the 'WDM node' (and not the 'DWDM node')

```suggestion
transponders are physically integrated within the same WDM node or
are physically located on a device external to the WDM node since it
both cases the optical transponders and the WDM node are under the
```

> control of the same O-PNC and abstracted as a single DWDM TE Node at the

Reviewer: I-D.ietf-ccamp-optical-impairment-topology-yang defines the 'WDM TE node' (and not the 'DWDM TE node')

```suggestion
control of the same O-PNC and abstracted as a single WDM TE Node at the
```

mechanisms which are outside the scope of this document, and reported
at the MPIs within the optical network topology.

In case of a multi-layer DWDM/OTN network domain, multi-layer

> intra-domain OTN links are supported by underlay DWDM tunnels: this

Reviewer: I-D.ietf-ccamp-wdm-tunnel-yang defines the 'WDM tunnel' model (and not the 'DWDM tunnel' model)

```suggestion
intra-domain OTN links are supported by underlay WDM tunnels: this
```

relationship is reported by the mechanisms described in
{{optical-path-discovery}}.

{: #optical-path-discovery}

## Optical Path Discovery

> The DWDM Tunnel Model is used to report all the DWDM tunnels

Reviewer: I-D.ietf-ccamp-wdm-tunnel-yang defines the 'WDM tunnel' model (and not the 'DWDM tunnel' model)

```suggestion
The WDM Tunnel Model is used to report all the WDM tunnels
```

The Ethernet client signal model and the Transparent CBR client
signal model are used to report all the connectivity services
provided by the underlay optical tunnels between Ethernet or CBR
client LTPs, depending on whether the connectivity service is frame-based

> or transparent. The underlay optical tunnels can be either DWDM

Reviewer: I-D.ietf-ccamp-wdm-tunnel-yang defines the 'WDM tunnel' model (and not the 'DWDM tunnel' model)

```suggestion
or transparent. The underlay optical tunnels can be either WDM
```

tunnels or, when the optional OTN switching layer is deployed, OTN
tunnels.

> The DWDM tunnels can be used to support either Ethernet or CBR client

Reviewer: I-D.ietf-ccamp-wdm-tunnel-yang defines the 'WDM tunnel' model (and not the 'DWDM tunnel' model)

```suggestion
The WDM tunnels can be used to support either Ethernet or CBR client
```

signals or multi-layer intra-domain OTN links. In the latter case,
the hierarchical-link container, defined in {{!I-D.ietf-teas-yang-te}},
associates

> the underlay DWDM tunnel with the supported multi-layer intra-domain

Reviewer: I-D.ietf-ccamp-wdm-tunnel-yang defines the 'WDM tunnel' model (and not the 'DWDM tunnel' model)

```suggestion
the underlay WDM tunnel with the supported multi-layer intra-domain
```

## Multi-technology IP Link Setup

As described in {{optical-path-computation}}, there is a one-to-one
relationship between a multi-technology intra-domain IP link and its
underlay optical tunnel.

Therefore, to set up a new multi-technology intra-domain IP link,
the MDSC requires the O-PNC to set up the optical tunnel (using either

> the DWDM Tunnel model or the OTN Tunnel model, if optional OTN switching

Reviewer: I-D.ietf-ccamp-wdm-tunnel-yang defines the 'WDM tunnel' model (and not the 'DWDM tunnel' model)

```suggestion
the WDM Tunnel model or the OTN Tunnel model, if optional OTN switching
```

created, as described above.

When member links are added or removed from a LAG link, the minimum
number of active member links required to consider the LAG link as up
may also need to be updated.

For example, with reference to {{fig-lag}}, the MDSC can request
P-PNC2 to add the multi-technology single-domain Ethernet link set up
between Ethernet LTP 3-1 on BR21 and Ethernet LTP 4-1 on P24 to the
existing Ethernet bundle link set up between Ethernet LTP 5-1 on node
BR21 and Ethernet LTP 6-1 on node P24.

After the LAG configuration has been updated, the P-PNC can also update
the bandwidth information of the multi-technology single-domain IP link
supported by the updated Ethernet bundled link.

{: #multi-technology-path-properties}

### Multi-technology TE path properties Configuration

The MDSC can discover the TE path properties (e.g., the list of
SRLGs, the delay) of a multi-technology IP link from the TE properties
of:

- the IP LTPs terminating the multi-technology IP link (e.g., the list of
SRLGs reported by the P-PNC using the packet TE topology model);

- the optical path (e.g., the list of SRLGs reported by the O-PNC

> using the DWDM or OTN tunnel model); and

Reviewer: I-D.ietf-ccamp-wdm-tunnel-yang defines the 'WDM tunnel' model (and not the 'DWDM tunnel' model)

```suggestion
using the WDM or OTN tunnel model); and
```

## L2/L3 VPN Network Service Setup

The MDSC can use the L2NM and L3NM network service models to request
the P-PNCs to setup L2/L3 VPN services, and the L2NM and L3NM TE
service mapping models to request the P-PNCs to configure the PE
routers to steer the L2/L3 VPN traffic to the selected TE tunnels
(e.g., MPLS-TE or SR-TE).

It is worth noting that the L2NM and L3NM TE service mapping models,
defined in {{?I-D.ietf-teas-te-service-mapping-yang}}, provide a list
of TE tunnel(s) that should be used to forward L2/L3 VPN traffic
between the two PEs terminating the listed TE tunnel(s). If the list
contains more than one TE tunnel for the same pair of PEs, these TE
tunnels are used to load balance the associated L2/L3 VPN traffic
between the same set of two PEs.

The possibility to request splitting the traffic between multiple TE
tunnels for the same PE pair in a way other than load balancing is
identified as a gap requiring further work and is outside the scope
of this draft.

{: #conclusions}

# Conclusions

The analysis provided in this document shows that the IETF YANG models

> described in 3.2 provide useful support for Packet Optical Integration

Reviewer: ```suggestion
described in {{yang}} provide useful support for Packet Optical Integration
```

- a mechanism to report client connectivity constraints imposed by
some muxponder design: this gap has been identified in {{muxponder}}.

Although not applicable to this document, it has been noted that being
able to use WSON and Flexi-grid topology models together (through
multi-inheritance) is not only useful for mixed fixed-grid and
flexible-grid DWDM network topologies but also the only viable option
for a mixed CWDM and DWDM network topology.

Although not applicable to this document, it has been noted that the

> DWDM tunnel model would also support optical tunnel setup in the case

Reviewer: I-D.ietf-ccamp-wdm-tunnel-yang defines the 'WDM tunnel' model (and not the 'DWDM tunnel' model)

```suggestion
WDM tunnel model would also support optical tunnel setup in the case
```

reconfigurable ROADM. Scenario here is to define only one port in the
routers and in the ROADM muxponder board at both ends as back-up
ports to recover any other port failure on client-side of the ROADM
(either on the IP router port side or on the muxponder side or on the
link
between them). When client-side port failure occurs, alarms are
raised to MDSC by IP-PNC and O-PNC (port status down, LOS etc.). MDSC
checks with OP-PNC(s) that there is no optical failure in the optical
layer.

There can be two cases here:

1. LAG was defined between the IP routers at the two ends. MDSC, after
checking

> that optical layer is fine between the two edge DWDM nodes, triggers
> the DWDM edge node re-configuration so that the IP router's back-up port

Reviewer: I-D.ietf-ccamp-optical-impairment-topology-yang defines the 'WDM node' (and not the 'DWDM node')

```suggestion
that optical layer is fine between the two edge WDM nodes, triggers
the WDM edge node re-configuration so that the IP router's back-up port
```

with its

> associated muxponder port can reuse the DWDM tunnel that was already in

Reviewer: I-D.ietf-ccamp-wdm-tunnel-yang defines the 'WDM tunnel' model (and not the 'DWDM tunnel' model)

```suggestion
associated muxponder port can reuse the WDM tunnel that was already in
```

use previously by the failed IP router port and adds the new link to
the LAG on the failure side.

   While the ROADM reconfiguration takes place, IP/MPLS traffic is
   using the reduced bandwidth of the IP link bundle, discarding
   lower priority traffic if required. Once back-up port has been

>    reconfigured to reuse the existing DWDM tunnel and the new link has

Reviewer: I-D.ietf-ccamp-wdm-tunnel-yang defines the 'WDM tunnel' model (and not the 'DWDM tunnel' model)

```suggestion
   reconfigured to reuse the existing WDM tunnel and the new link has
```

the same time MDSC, after checking that optical network connection
is still fine, would trigger the reconfiguration of the back-up
port of the IP router and of the muxponder to re-use the same

> DWDM tunnel as the one used originally for the failed IP router port. Once

Reviewer: I-D.ietf-ccamp-wdm-tunnel-yang defines the 'WDM tunnel' model (and not the 'DWDM tunnel' model)

```suggestion
WDM tunnel as the one used originally for the failed IP router port. Once
```

everything has been correctly configured, MDSC Global PCE could
suggest to the operator to trigger a possible re-optimization of
the back-up MPLS path to go back to the  MPLS primary path through

> the back-up port of the IP router and the original DWDM tunnel if overall

Reviewer: I-D.ietf-ccamp-wdm-tunnel-yang defines the 'WDM tunnel' model (and not the 'DWDM tunnel' model)

```suggestion
the back-up port of the IP router and the original WDM tunnel if overall
```

modulation format, the FEC etc.

The controller, when asked to set up a client connectivity service,

> needs to find a DWDM tunnel suitable to comply the DWDM port

Reviewer: I-D.ietf-ccamp-wdm-tunnel-yang defines the 'WDM tunnel' model (and not the 'DWDM tunnel' model)

```suggestion
needs to find a WDM tunnel suitable to comply the DWDM port
```

The controller, when asked to set a 10GE client connectivity service
between two muxponder's client ports, needs first to check whether

> there is already an existing DWDM tunnel between the two muxponders

Reviewer: I-D.ietf-ccamp-wdm-tunnel-yang defines the 'WDM tunnel' model (and not the 'DWDM tunnel' model)

```suggestion
there is already an existing WDM tunnel between the two muxponders
```

and then take different actions:

> 1. if the DWDM tunnel already exists, the controller needs only to

Reviewer: I-D.ietf-ccamp-wdm-tunnel-yang defines the 'WDM tunnel' model (and not the 'DWDM tunnel' model)

```suggestion
1. if the WDM tunnel already exists, the controller needs only to
```

enable the 10GE client ports to establish the 10GE client
connectivity service;

> 2. if the DWDM tunnel does not exist, the controller has to first
> establish the DWDM tunnel, finding a proper optical path matching

Reviewer: I-D.ietf-ccamp-wdm-tunnel-yang defines the 'WDM tunnel' model (and not the 'DWDM tunnel' model)

```suggestion
2. if the WDM tunnel does not exist, the controller has to first
establish the WDM tunnel, finding a proper optical path matching
```

the optical parameters of the two muxponders' trunk ports (e.g.,
an OTSi carrying an OTU4), and then enable the 10GE client ports
to establish the 10GE client connectivity service.

> Since multiple client connectivity services are sharing the same DWDM

Reviewer: I-D.ietf-ccamp-wdm-tunnel-yang defines the 'WDM tunnel' model (and not the 'DWDM tunnel' model)

```suggestion
Since multiple client connectivity services are sharing the same WDM
```

In case of flexible configuration, since the two muxponders are under
the control of the same O-PNC, the configuration of the multiplexing
label, regardless of whether it is a standard or vendor-specific
label, can be done by the O-PNC using mechanisms which are
vendor-specific and outside the scope of this document. The MDSC can just

> request the O-PNC to setup a client connectivity service over a DWDM

Reviewer: I-D.ietf-ccamp-wdm-tunnel-yang defines the 'WDM tunnel' model (and not the 'DWDM tunnel' model)

```suggestion
request the O-PNC to setup a client connectivity service over a WDM
```
